### ISSUE TITLE
API Documentation

### ISSUE THREAD

**Issue by DungChau1996** [eclipse-cyclonedds/cyclonedds#423]
As I know, so far, the only documentation I can find is the CycloneDDS-0.1.0 ( cannot find 0.5.0) which has API reference. To be honest, it is hard to follow. 

My question is that: Is there any more completed version of API documentation? Or Doxygen page? If not, is there any plan to release those documentation and also the Release version 0.5.0?

**Comment by trickapm** [comment]
There are certainly plans to do automated documentation builds, hosted on readthedocs, for each release. The API reference will be generated using Doxygen, transformed into html by Sphinx. As you've noticed, the documentation is fairly sparse and needs improvements, both the content and the doxygen tags/syntax. It will probably not be published until deemed sufficiently complete and usable as a CycloneDDS Manual to new users. I've forked the repo intending to work on this but haven't made much progress yet.

Meanwhile you should be able to build docs manually by enabling the BUILD_DOCS cmake option (or just view the docs in dds.h directly). Contributions to improve docs are appreciated and will not be lost as doxygen will remain the mechanism used to generate API reference.

**Comment by thijsmie** [comment]
The API docs are available on [cyclonedds.io](https://cyclonedds.io) for some time now, go take a look!
